### PR TITLE
conf: replace Kconfig defaults with board-based conf file settings

### PR DESCRIPTION
**Pull request by mike-scott**
Kconfig defconfig settings seem like a good idea on the surface, but
the reality is that they skip all dependency and validation checks.
Much better to set the defaults per board using board-based conf files
which are included automatically if they exist.

This patch has no user process / documentation changes.  I also compared
the output of outdir/*/.config for all 3 targets affected by this patch.
They were identical before and after.

Signed-off-by: Michael Scott <michael.scott@linaro.org>

### CODE REVIEW

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
Application Build Succeeded.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

File: boards/96b_carbon.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+CONFIG_BLUETOOTH_CONTROLLER_TX_BUFFERS=7
+CONFIG_BLUETOOTH_RX_STACK_SIZE=2048
+
+# HW settings
+CONFIG_TEXT_SECTION_OFFSET=0x100
+CONFIG_FLASH_LOAD_OFFSET=0x20000
+
+# temp settings
+CONFIG_TEMP_STM32F401X_NAME="fota-mcu-temp"
+
```
Comment (mbolivar) [inline boards/96b_carbon.conf]: Extra whitespace at end of file.

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

**Comment by gatek33p3r** [comment]
Application Build Succeeded.

**Comment by mike-scott** [comment]
V2: removed extra white space from new board/*.conf files.

**Pull request merged**